docs: changelog 2025-10-24 (#4063)
neondatabase/website: content/changelog/2025-10-24.md

---
title: AI tooling, default storage expansion, and Postgres updates
---

## Claude Code plugin for Neon

We've launched a new plugin that brings Neon's capabilities directly into Claude Code. The new plugin includes:

- **Claude Skills** to streamline key workflows:
  - **neon-drizzle**: Set up Drizzle ORM with Neon
  - **neon-serverless**: Configure connections with Neon's serverless Postgres driver
  - **neon-toolkit**: Manage databases, projects, and branches using the Neon API
  - **add-neon-knowledge**: Access Neon documentation snippets and usage examples

- **Neon MCP server integration** that lets Claude interact with Neon in real time to query projects, manage databases and branches, and run SQL or migrations

- **Context rules (.mdc files)** that can be used in other AI tools like Cursor

> Install it from the Claude plugin marketplace:

Author: ```suggestion
Install it from our marketplace:
```

---
title: AI tooling, storage expansion, and Postgres updates
---

### Claude Code plugin for Neon

We've launched the **Neon Claude Code plugin**, an integration that brings Neon's capabilities directly into Claude Code, Anthropic's AI development environment.

The plugin includes:

- **4 Claude Skills** for guided workflows:
  - **neon-drizzle**: Set up Drizzle ORM with Neon, including schema creation and connection setup
  - **neon-serverless**: Configure Neon's serverless Postgres driver and test connections
  - **neon-toolkit**: Use the Neon Management API to create databases, projects, and branches dynamically
  - **add-neon-knowledge**: Access Neon documentation snippets and usage examples

- **MCP server integration** that lets Claude interact with Neon's live API endpoints to:
  - Query Neon for project information
  - Create or delete branches and databases
  - Validate connection strings
  - Run SQL queries and migrations

> - **Portable context rules (.mdc files)** that can be used in other AI tools like Cursor

Author: ```suggestion
- **Context rules (.mdc files)** that can be used in other AI tools like Cursor
```

## Get started with Neon + Cursor

We've introduced a new init command to help you quickly connect your app and enable the Neon integration in your Cursor chat. Run the following command in your app's root directory:

```bash
npx neonctl init
```

The command walks you through an interactive setup:

```
🚀 Neon Project Initialization

Step 1/3: Configuring Neon MCP Server...
Step 2/3: Creating neon.md with detailed guidelines...
Step 3/3: Creating AGENTS.md for Cursor...

✓ Success! Neon project initialized.
```

After setup, you can ask your Cursor chat to "Get started with Neon" and it will have full context about your project and Neon best practices.

This feature is currently in beta for Cursor, with VS Code and Claude Code support coming soon.

## MCP server: Schema diff and migration generation

Our MCP server now supports schema diff generation and zero-downtime migration creation. Ask your AI assistant:

```
Can you generate a schema diff for branch br-feature-auth in project my-app?
```

The assistant will compare the branch schema with its parent, show what changed, and offer to generate a zero-downtime migration to apply those changes to the parent branch.

> This makes it easier to develop schema changes on feature branches and promote them to production. For more information, see [Neon MCP Server](/docs/ai/neon-mcp-server).

Author: ```suggestion
This makes it easier to develop schema changes on feature branches and promote them when ready. For more information, see [Neon MCP Server](/docs/ai/neon-mcp-server).
```

## Storage quota doubled to 16TB

> We've doubled our default storage quota from 8TB to 16TB. This means you can now run databases up to 16TB without contacting us. If you need to run larger databases, please [reach out to our team](https://neon.tech/contact-sales).

Author: ```suggestion
We've doubled our default storage quota from 8TB to 16TB. This means you can now run databases up to 16TB without contacting us to increase your limit. If you need to run larger databases, please [reach out to our team](https://neon.tech/contact-sales).
```

---
title: AI tooling, default storage expansion, and Postgres updates
---

> ### Claude Code plugin for Neon

Author: ```suggestion
## Claude Code plugin for Neon
```

| Extension | Old Version | New Version |
| :-------- | :---------- | :---------- |
| anon      | 2.1.0       | 2.4.1       |

To upgrade from a previous version of an extension, follow the instructions in [Update an extension version](/docs/extensions/pg-extensions#update-an-extension-version).

For a complete list of Postgres extensions supported by Neon, see [Postgres extensions](/docs/extensions/pg-extensions).

<details>
<summary>**Fixes & improvements**</summary>

- **Child branch storage now capped at logical data size**

  We've introduced a storage billing cap for child branches. Previously, child branch storage cost was based on all data changes over time. Now, you're billed for the minimum of accumulated changes or your actual data size, ensuring you never pay more than the logical size of your data on a child branch. This change makes child branch storage costs more predictable and helps avoid charges from long-lived branches.

- **Neon Launchpad packages renamed**

  Neon Launchpad's packages were renamed: `neondb` → `get-db` (CLI) and `vite-plugin-postgres` → `vite-plugin-db` (Vite plugin). Use `npx get-db` to initiate Neon Launchpad.
  

> _Neon Launchpad enables instant provisioning of a Postgres database without configuration or account creation. See [Neon Launchpad](/docs/reference/neon-launchpad) to learn more._

Author: ```suggestion
    _Neon Launchpad enables instant provisioning of a Postgres database without configuration or account creation. See [Neon Launchpad](/docs/reference/neon-launchpad) to learn more._
```
